src/fetchApi.ts

```diff
@@ -1,9 +1,37 @@
-export const fetchArtworks = async () => {
+export const fetchArtworks = async (
+  isOnView: boolean,
+  isPublicDomain: boolean,
+  isHiddenGem: boolean
+) => {
   try {
-    const response = await fetch(
-      "https://api.artic.edu/api/v1/artworks?page=1&limit=10&fields=title,artist_title,place_of_origin,date_display,image_id"
-    );
+    const query = {
+      query: {
+        bool: {
+          must: [
+            ...(isOnView ? [{ term: { is_on_view: true } }] : []),
+            ...(isPublicDomain ? [{ term: { is_public_domain: true } }] : []),
+            ...(isHiddenGem
+              ? [{ term: { has_not_been_viewed_much: true } }]
+              : []),
+          ],
+        },
+      },
 
+      limit: 10,
+
+      fields: [
+        "title",
+        "artist_title",
+        "place_of_origin",
+        "date_display",
+        "image_id",
+      ],
+    };
+
+    const params = encodeURIComponent(JSON.stringify(query));
+    const url = `https://api.artic.edu/api/v1/artworks/search?params=${params}`;
```

Comment:
Ah, I found it now!

> Our API accepts queries through both `GET` and `POST`. It might be easier to visualize how the `GET` query above is structured by comparing it to its `POST` equivalent:
> ```shell
> curl --location --request POST 'https://api.artic.edu/api/v1/artworks/search' \
> --header 'Content-Type: application/json' \
> --data-raw '{
>     "q": "cats",
>     "query": {
>         "term": {
>             "is_public_domain": true
>         }
>     }
> }'
> ```
> For production use, we recommend using `GET` and passing the entire query as minified URL-encoded JSON via the `params` parameter.

So, using `POST` was actually valid! I'm sorry, I didn't know about this before.

But as they're recommending `GET` switching from `POST` to `GET` seems to be the better choice.

And using `params` is valid as well

Reply:
well i now know how to do both! 🙌